File: src/main/java/a_basic/AssignmentOperators.java
package a_basic;

/**
 * 赋值运算符
 */
public class AssignmentOperators {
    public static void main(String[] args) {
        int a = 10;  // 基本的赋值运运算符
        System.out.println("***************************");

        // +=,-=,/=,*=,%=;(自动进行强制类型转换)
        a += 20.34; // 等价于a = (int)(a + 20.34)
        System.out.println(a);
        System.out.println("***************************");

        short s = 2;
        // s = s + 1;  // 报错:1是int类型数据,故s+1也是int类型数据
        /*
        解决方法如下:
        法1, s = (short)(s+1);
        法2, s += 1;
        */
    }
}

File: src/main/scala/sa_basic/AssignmentOperators.scala
package sa_basic

/**
 * 赋值运算符
 */
object AssignmentOperators {
    def main(args: Array[String]): Unit = {
        var a = 10 // 基本的赋值运运算符
        // println("***************************")
        // a += 20 // +=,-=,/=,*=,%=;注意:Scala中没有自加(++),自减(--)操作符
        // println(a)
        // println("***************************")
        var s: Short = 2
        // s = s + 1;  // 报错:1是Int类型数据,故s+1也是Int类型数据
        /*
        解决方法如下:
        s = (s + 1).toShort
        */
    }
}
